File: python/src/exercise21.py
def fib(n: int) -> int:
    match n:
        case 0:
            return 0
        case 1:
            return 1
        case _ if 1 < n:
            return fib(n-2) + fib(n-1)
        case _:
            raise ValueError(n)

File: rust/src/exercise21.rs
pub fn fib(n: i32) -> i32 {
    match n {
        0 => 0,
        1 => 1,
        _ if 1 < n => fib(n - 2) + fib(n - 1),
        _ => panic!("{}", n),
    }
}
